src/transformers/modeling_utils.py

```diff
@@ -2422,7 +2422,8 @@ def _fix_key(key):
 
         if remove_prefix_from_model:
             expected_keys_not_prefixed = [s for s in expected_keys if not s.startswith(prefix)]
-            expected_keys = [".".join(s.split(".")[1:]) if s.startswith(prefix) else s for s in expected_keys]
+            _prefix = f"{prefix}."
+            expected_keys = [".".join(s.split(".")[1:]) if s.startswith(_prefix) else s for s in expected_keys]
```

Comment:
This would fail for `esm`/`esmfold` since `esm` is a prefix for `esmfold` it would remove `emsfold.` prefix instead of the intended `esm.`

This was silent before, but with the new test addition the following modification would fail to get `key[-1]` when trying to identify missing keys.

Reply:
Seems like the right fix, thanks!

Reply:
Ok, there was a second place to update the fix to make the `splinter` test pass.

Reply:
https://github.com/huggingface/transformers/pull/20042/files#diff-6b72b98c4c2dcfc6cc606843917733f5d858374fbc22a735ff483bbc0c1e63eaR2655